pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -420,19 +420,39 @@ func NewJobController(config *rest.Config, serverOption *options.ServerOption) *
 func (qjm *XController) PreemptQueueJobs() {
 	qjobs := qjm.GetQueueJobsEligibleForPreemption()
 	for _, q := range qjobs {
-		newjob, e := qjm.queueJobLister.AppWrappers(q.Namespace).Get(q.Name)
-		if e != nil {
-			continue
-		}
-		newjob.Status.CanRun = false
+		//If pods failed scheduling generate new preempt condition
+		//ignore co-scheduler failed scheduling events.
+		if q.Status.PendingPodsFailedSchd > 0 {
+			newjob, e := qjm.queueJobLister.AppWrappers(q.Namespace).Get(q.Name)
+			if e != nil {
+				continue
+			}
+			newjob.Status.CanRun = false
 
-		message := fmt.Sprintf("Insufficient number of Running pods, minimum=%d, running=%v.", q.Spec.SchedSpec.MinAvailable, q.Status.Running)
-		cond := GenerateAppWrapperCondition(arbv1.AppWrapperCondPreemptCandidate, v1.ConditionTrue, "MinPodsNotRunning", message)
-		newjob.Status.Conditions = append(newjob.Status.Conditions, cond)
+			message := fmt.Sprintf("Pods failed scheduling failed=%v, running=%v.", q.Status.PendingPodsFailedSchd, q.Status.Running)
+			cond := GenerateAppWrapperCondition(arbv1.AppWrapperCondPreemptCandidate, v1.ConditionTrue, "PodsFailedScheduling", message)
```

Comment:
Suggest checking for duplicate conditions to improve performance of job updates by not appending duplicates.  See usage examples for utility fn `isLastConditionDuplicate()`

Reply:
Done